src/hu/elte/txtuml/stdlib/net/URLImpl.java

```diff
@@ -0,0 +1,339 @@
+     * the scheme of the context, then the new URL is created as an absolute
+     * URL based on the spec alone. Otherwise the scheme component is inherited
+     * from the context URL.
+     * <p>
+     * If the authority component is present in the spec then the spec is
+     * treated as absolute and the spec authority and path will replace the
+     * context authority and path. If the authority component is absent in the
+     * spec then the authority of the new URL will be inherited from the
+     * context.
+     * <p>
+     * If the spec's path component begins with a slash character
+     * &quot;/&quot; then the
+     * path is treated as absolute and the spec path replaces the context path.
+     * <p>
+     * Otherwise, the path is treated as a relative path and is appended to the
+     * context path, as described in RFC2396. Also, in this case,
+     * the path is canonicalized through the removal of directory
+     * changes made by occurrences of &quot;..&quot; and &quot;.&quot;.
+     * <p>
+     * For a more detailed description of URL parsing, refer to RFC2396.
+     *
+     * @param      context   the context in which to parse the specification.
+     * @param      spec      the {@code String} to parse as a URL.
+     * @exception  MalformedURLException  if no protocol is specified, or an
+     *               unknown protocol is found, or {@code spec} is {@code null}.
+     * @see        java.net.URL#URL(java.lang.String, java.lang.String,
+     *                  int, java.lang.String)
+     * @see        java.net.URLStreamHandler
+     * @see        java.net.URLStreamHandler#parseURL(java.net.URL,
+     *                  java.lang.String, int, int)
+     */
+	public URLImpl(hu.elte.txtuml.stdlib.net.URL context, String spec) throws MalformedURLException {
+		this.instance = new java.net.URL(context.toURL(), spec);
+	}
+	
+    /**
+     * Compares this URL for equality with another object.<p>
+     *
+     * If the given object is not a URL then this method immediately returns
+     * {@code false}.<p>
+     *
+     * Two URL objects are equal if they have the same protocol, reference
+     * equivalent hosts, have the same port number on the host, and the same
+     * file and fragment of the file.<p>
+     *
+     * Two hosts are considered equivalent if both host names can be resolved
+     * into the same IP addresses; else if either host name can't be
+     * resolved, the host names must be equal without regard to case; or both
+     * host names equal to null.<p>
+     *
+     * Since hosts comparison requires name resolution, this operation is a
+     * blocking operation. <p>
+     *
+     * Note: The defined behavior for {@code equals} is known to
+     * be inconsistent with virtual hosting in HTTP.
+     *
+     * @param   obj   the URL to compare against.
+     * @return  {@code true} if the objects are the same;
+     *          {@code false} otherwise.
+     */
+    public boolean equals(Object obj) {
```

Comment:
`public boolean equals(Object ob)` is a method of `java.lang.Object` which I override to make comparison work properly.  It is not logical part of `URL` interface, because of that I intentionally didn't put it there. Even if we add it to the interface, the compiler will not complain as it already inherits `equals(Object ob)` from the base `java.lang.Object` class.

It is important that `equals` takes an `Object`. Otherwise, unexpected behavior occurs. Consider the following scenario if we take `URL` as an argument instead of `Object`.
```
URL uriInstance1 = new URL("http://google.com");
URL uriInstance2 = new URL("http://google.com");
Object objInstance = new URL("http://google.com");

uriInstance1.equals(uriInstance2); //returns true
uriInstance1.equals(objInstance); // returns false, because URI.equals(Object) is inherited from java.lang.Object 
```

In addition, since `java.lang.Object` is a superclass of the all Java classes, in my opinion, it should not make any problem. Please let me know if you agree.